scripts/build/termux_step_install_license.sh

```diff
@@ -52,35 +53,41 @@ termux_step_install_license() {
 				'BSD'|'BSD 2-Clause'|'BSD 3-Clause'|'BSD Simplified'\
 				|'curl'|'HPND'|'ISC'|'Libpng'|'MIT'|'Openfont-1.1'\
 				|'PythonPL'|'X11'|'ZLIB')
-				# Find the license file in the source files
+				# We only want to include the license files from the source files once
+				(( FROM_SOURCES )) || \
 				for FILE in "${COMMON_LICENSE_FILES[@]}"; do
+				# Find the license file(s) in the source files
 					[[ -f "$TERMUX_PKG_SRCDIR/$FILE" ]] && {
 						if (( COUNTER )); then
 							cp -f "${TERMUX_PKG_SRCDIR}/$FILE" "${TERMUX_PREFIX}/share/doc/${TERMUX_PKG_NAME}/copyright.${COUNTER}"
 						else
 							cp -f "${TERMUX_PKG_SRCDIR}/$FILE" "${TERMUX_PREFIX}/share/doc/${TERMUX_PKG_NAME}/copyright"
 						fi
-						# since this is a post-increment, (( 0 )) would be falsey
-						# thus `set -e` would kill the script on the first iteration
-						# using `true` prevents this
-						: $(( COUNTER++ ))
+						(( ++COUNTER, ++FROM_SOURCES ))
 					}
 				done
+				if (( ! FROM_SOURCES )); then
+					# If we get here no license file could be found
+					termux_error_exit "${TERMUX_PKG_NAME}: Could not find a license file for $LICENSE in the package sources"
```

Comment:
Add a comma `,` after `here` and in below comment too.

Reply:
Actually I think explaining the second `if (( ! FROM_SOURCES )); then` works better here.

Reply:
You need to move the second `if (( ! FROM_SOURCES )); then` inside the first one, otherwise when second non-generic license is processed, the second if will get triggered and abort.

And use "If we have no**t** found".

Reply:
Oh yeah you're right it would need to be defined outside the while loop otherwise it gets reset for each license.

Reply:
>the second if will get triggered and abort.

Sorry, it won't abort, but wrong place for it. The second abort `if (( ! FROM_SOURCES ))` should be inside the first `if (( ! FROM_SOURCES ))` in which licenses are actually searched so that it isn't run after first non-generic license loop.

And now you have moved the `FROM_SOURCES` initialization to the wrong `if` (`TERMUX_PKG_LICENSE_FILE`).

Reply:
Yep got it.
Maybe I should have gotten more than 3 hours of sleep.

Reply:
ما هوا الخطء

Reply:
1. Move `local FROM_SOURCES=0` along with `local TO_LICENSE` inside the `else`.
2. The `FILE` variable should be defined as `local` that's used in the `COMMON_LICENSE_FILES` for loop.
3. The `# Find the license file(s) in the source files` comment should be above the for loop.

Otherwise looks fine.

> Maybe I should have gotten more than 3 hours of sleep.

Most definitely, brain gets really affected with poor sleep, which one may not even realize. And I am sure your editor environment is to blame as well :p

Reply:
Rebasing the PR and working on it now.
Hopefully this will be the last revision.

Reply:
- `local FROM_SOURCES=0` is in the `else`
- `FILE` has been declared local.
- loop comment is now above the loop.

Think that's all.

Reply:
Looks good now. Thanks.

One thing though, the code under the two case statements `'BSD'|...)` and `*)` should be indented except the `;;` at the end of each, looks confusing currently. But you can merge it regardless.

Reply:
Eh we've done like 20 revisions, what's one more.

Reply:
lolz, true, do it then. I should have caught it earlier.

Reply:
Just pushed it in the latest revision

Reply:
Great! ~~Now let's wait for the next production break~~

Reply:
<del>Does not changing the commit message to actually reflect the final state of the PR count?</del>

Reply:
~~Wrong commit message is a getting-kicked-out-of-org offense!~~